Library/Homebrew/download_strategy.rb

```diff
@@ -421,9 +433,12 @@ def mirrors
 
   def resolve_url_and_basename(url)
```

Comment:
Would be good to adjust the name to clarify this also returns a time. I think ideally this would now return a hash to improve readability; `resolve_url_and_basename(url)[0..1]` is a bit opaque.